clients/lsp-clangd.el

```diff
@@ -178,6 +178,49 @@ This must be set only once after loading the clang client.")
   :risky t
   :type '(repeat string))
 
+(defun lsp-clangd-find-compile-commands-dir ()
+  "Return the full directory path (or nil) that contain a compile_commands.json.
+
+If multiple compile_commands.json files are
+found in the project directory tree - ask the user to choose."
+  ;; TODO lsp-workspace-root won't find anything at the start of the workspace
+  (let ((candidate-dirs (directory-files-recursively (lsp-workspace-root) "compile_commands.json")))
+    (pcase (length candidate-dirs)
+          ;; TODO can I lsp--warn-user-in-message-and-log at the same time
+      (`0 (lsp--warn "Cannot find compile_commands.json - intellisense might not work")
+          nil)
+      ;; if you find 1 only - assume it's correct and strip the dirname from the full path
+      ;; happy path
+      (`1 (file-name-directory (car candidate-dirs)))
+      ;; if you have several - ask the user to choose with lsp--completing-read
+      ;; TODO if Ctrl-G from inside this completing read still falls to default lsp-clients-clangd-args
+      (_ (file-name-directory (lsp--completing-read
+           "Found compile_commands.json in several directories, which one do you want to use?"
```

Comment:
Nit: I suggest mentioning that you may use lsp-clients-clangd-args to avoid being asked each time you open that project.